### PR TITLE
[Cherry-Pick] Remove arrow uasge in FieldData

### PR DESCRIPTION
**Pull request by xige-16**
Cherry pick from https://github.com/milvus-io/milvus/pull/21523
fix issue: https://github.com/milvus-io/milvus/issues/20995
/kind improvement

Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

**Comment by xige-16** [comment]
rerun ut

File: internal/core/src/storage/FieldDataFactory.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        case DataType::STRING:
+        case DataType::VARCHAR:
+            return std::make_shared<FieldData<std::string>>(type);
+        case DataType::VECTOR_FLOAT:
+            return std::make_shared<FieldData<FloatVector>>(dim, type);
+        case DataType::VECTOR_BINARY:
+            return std::make_shared<FieldData<BinaryVector>>(dim, type);
+        default:
+            throw NotSupportedDataTypeException(
+                GetName() + "::CreateFieldData" + " not support data type " +
```
Comment (jiaoew1991) [inline internal/core/src/storage/FieldDataFactory.cpp:48]: use fmt::format to prettify it

Comment (xige-16) [reply internal/core/src/storage/FieldDataFactory.cpp:48]: next pr

**Pull request merged**